### DISCUSSION TITLE
what is the maximum entity count, variable count and approximate value count the TimeFold can solve?

### DISCUSSION THREAD

**Discussion by yush-sys** [TimefoldAI/timefold-solver#1011] [General]
I found the following line in the TimeFold documentation "Timefold Solver is stable, reliable and scalable. It has been heavily tested with
unit, integration, and stress tests, and is used in production throughout the world. One example handles over 50 000 variables with 5000
values each, multiple constraint types and billions of possible constraint matches. We offer two editions of Timefold Solver." 
        But in my application in have a problem scale of entity count: 4976, variable count: 9952, approximate value count: 2166 and
approximate problem scale: 2.591009 x 10^25454. and it didn't solve. I let the solver run for about 4 hrs. I am currently using community
edition. Do I have to get the Enterprise edition for this problem scale?

**Comment by triceo** [comment]
Hello @yush-sys - what do you mean by "did not solve"?
In general, your problem is large enough to benefit from an Enterprise edition, but there is no reason why the Community Edition would refuse to solve any problem.

**Reply by yush-sys** [reply]
I have the local search to terminate when the solver stops improving the score.
termination
  unimprovedSecondsSpentLimit 20 unimprovedSecondsSpentLimit
termination
after letting it run for 4 hrs. it still didn't return any result.

**Reply by triceo** [reply]
That's good, isn't it? The solver is still improving. (And if you let it, on these large data sets, it will likely keep improving for days.) Let it run for as much time as you have, and then take whatever best solution the solver has reached so far. It is not reasonable to expect that, for these massive data sets, you will exhaust the solver's potential in minutes or even hours.

Enterprise Edition is going to be very helpful in this case; it is faster, and if your problem has a spatial aspect to it, Nearby Selection will bring a massive boost to solution quality.
